### PR TITLE
fix: Bump teslajsonpy to 3.12.2

### PR DESCRIPTION
**Pull request by ehendrix23**
Bump teslajsonpy to 3.12.2 which includes fix for API service

### CODE REVIEW

**Comment by alandtse** [comment]
Please review other bumps. Other files will also also need to be changed.

**Comment by ehendrix23** [comment]
Updated the additional files. Not sure why the test would fail though as nothing there has been changed.

**Comment by alandtse** [comment]
Someone probably broke the test with a commit.

**Comment by alandtse** [comment]
Thanks for figuring that out. Is the other PR #1069 a prereq of this? Normally I'd want to keep the test fix as a separate PR so the bump PR is very clean.

**Comment by ehendrix23** [comment]
> Thanks for figuring that out. Is the other PR #1069 a prereq of this? Normally I'd want to keep the test fix as a separate PR so the bump PR is very clean.

Nope, no dependencies. PR #1069 was purely to update for HA 2024.11 and had nothing to do with teslajsonpy. :-)

FYI, seems I screwed some things up so closing this PR for the bump and creating a new one instead that is clean. :-)

File: custom_components/tesla_custom/services.py [hunk 1]
```diff
@@ -5,11 +5,11 @@
 
 import logging
 
+import voluptuous as vol
```
Comment (alandtse) [inline custom_components/tesla_custom/services.py:8]: Did black do this?

Comment (ehendrix23) [reply custom_components/tesla_custom/services.py:8]: I would say yes, I didn't add it. :-)

**Pull request closed without merging**